tests/backup/backup_helper.go

```diff
@@ -1181,13 +1247,128 @@ func backupSuccessCheck(backupName string, orgID string, retryDuration int, retr
+		Uid:   bkpUid,
+		OrgId: orgID,
+	}
+	resp, err := backupDriver.InspectBackup(ctx, backupInspectRequest)
+	if err != nil {
+		return false, nil, err
+	}
+	backupStatus := (resp.GetBackup().GetStatus().Status == api.BackupInfo_StatusInfo_Success)
+	log.Infof("Backup [%s] created successfully", backupName)
+	return backupStatus, resp, nil
+}
+
+func ValidateBackup(backupInspectResponse *api.BackupInspectResponse, clusterAppsContexts []*scheduler.Context) (*BackupRestoreContext, error) {
+	bkpObjs := backupInspectResponse.Backup.Resources
+	if backupSpecContext, err := GetBackupSpecObjectsContexts(bkpObjs, clusterAppsContexts); err != nil {
+		return nil, fmt.Errorf("GetBackupSpecObjects Err: %v", err)
+	} else {
+		return &BackupRestoreContext{
+			schedCtxs: backupSpecContext,
+		}, nil
+	}
+}
+
+// Filters the source namespaces' contexts to only include backup objects
+// MAKE SURE to include Contexts of ALL namespaces which contain the backup objects
+func GetBackupSpecObjectsContexts(resourceInfos []*api.ResourceInfo, clusterAppsContexts []*scheduler.Context) ([]*scheduler.Context, error) {
+	log.InfoD("Validation: Getting the backup objects (specs) from the Context")
+
+	var backupAppContexts []*scheduler.Context = make([]*scheduler.Context, 0)
+	var allBkpObjs []*api.ResourceInfo = make([]*api.ResourceInfo, 0)
+
+	//for each clusterAppsContext (namespace), we return the corresponding BackupSpecObject
+	for _, clusterAppsContext := range clusterAppsContexts {
+		var specObjects []interface{} = make([]interface{}, 0)
+		// the Backup objects which were found in this namespace
+		var resInfoObjs []*api.ResourceInfo = make([]*api.ResourceInfo, 0)
+
+		for _, resource := range resourceInfos {
+			if resource.Namespace == clusterAppsContext.ScheduleOptions.Namespace {
+				resInfoObjs = append(resInfoObjs, resource)
+			} else if resource.Namespace == "" {
+				// IF it is therein this context, we'll work with it
+				resInfoObjs = append(resInfoObjs, resource)
+			}
+		}
+
+	specloop:
+		for _, spec := range clusterAppsContext.App.SpecList {
+			name, kind, ns, err := GetSpecNameKindNamepace(spec)
+			if err != nil {
+				log.Error(err)
+				continue specloop
+			}
+			if name != "" /*&& kind != ""*/ /*kind is empty for certain objects :(*/ {
+				for _, bkpObj := range resInfoObjs {
+					if name == bkpObj.Name /*&& kind == bkpObj.Kind*/ {
+						clone := spec
+						// The spec for the object/resource was found in the Context (namespace). Appending it.
+						if ns == "" {
+							log.InfoD("The resource(name: %s, kind: %s, with NO NAMESPACE value) was found in the context '%s'", name, kind, clusterAppsContext.ScheduleOptions.Namespace)
+						}
+						specObjects = append(specObjects, clone)
+						continue specloop
+					}
+				}
+			} else {
+				log.Warnf("some error with getting GetSpecNameKindNamepace with Spec Name: %s, Kind: %s, Namespace: %s, in local Context (NS): %s", name, kind, ns, clusterAppsContext.ScheduleOptions.Namespace)
```

Comment:
why there will be some error, if there is an error handle it with proper message

Reply:
✅